### ISSUE TITLE
Request Body Sometimes Ignored on Blueprint Routes

### ISSUE THREAD

**Issue by jamesstidard** [sanic-org/sanic#2096]
I have a request handler that will sometime fail a unit test and sometimes succeed. It's failing as sometimes the request body is empty on being handed to the handler. This appears to only occur sometimes as it matters the order of registered handlers under the same Blueprint and those handlers are stored in a set within the Blueprint (from what I can tell).

I originally opened the issue over here: https://github.com/sanic-org/sanic-testing/issues/14 as I thought it was specific to the testing framework, before I dug a little deeper.

Please read that ticket for example code, versions, and more detail on where the issue is arising from.

Thanks

**Comment by brianmcburney** [comment]
I notice this happening if routes have the same path for different methods. With the below routes, in the post, the body is (almost?) always empty.
```
@app.get("/")
def get_method(request):
    ...

@app.post("/")
def post_method(request):
    assert request.json is not None
```

**Comment by ahopkins** [comment]
@brianmcburney Can you let me know what version of `sanic-routing` is installed?

```
$ python -c 'import sanic_routing;print(sanic_routing.__version__)'
```

**Comment by brianmcburney** [comment]
`0.5.2`

If it would be helpful I can share my entire (runnable) test project.

**Comment by Tronic** [comment]
Probably a duplicate of #2103

**Comment by Tronic** [comment]
@brianmcburney As a workaround, you should be able to add `await request.receive_body()` as the first line in your handler.

**Comment by brianmcburney** [comment]
@Tronic that does fix it, thanks

**Comment by Tronic** [comment]
Closing as this bug has been triaged and the fix is being worked on in #2103. Thanks for reporting!
